server/service/mdm_test.go

```diff
@@ -947,3 +947,389 @@ func TestMDMWindowsConfigProfileAuthz(t *testing.T) {
+			true,
+			ptr.Uint(1),
+			nil,
+			nil,
+			"",
+		},
+		{
+			"team maintainer, DOES NOT belong to team",
+			&fleet.User{Teams: []fleet.UserTeam{{Team: fleet.Team{ID: 2}, Role: fleet.RoleMaintainer}}},
+			true,
+			ptr.Uint(1),
+			nil,
+			nil,
+			authz.ForbiddenErrorMessage,
+		},
+		{
+			"team observer, DOES belong to team",
+			&fleet.User{Teams: []fleet.UserTeam{{Team: fleet.Team{ID: 1}, Role: fleet.RoleObserver}}},
+			true,
+			ptr.Uint(1),
+			nil,
+			nil,
+			authz.ForbiddenErrorMessage,
+		},
+		{
+			"team observer, DOES NOT belong to team",
+			&fleet.User{Teams: []fleet.UserTeam{{Team: fleet.Team{ID: 2}, Role: fleet.RoleObserver}}},
+			true,
+			ptr.Uint(1),
+			nil,
+			nil,
+			authz.ForbiddenErrorMessage,
+		},
+		{
+			"user no roles",
+			&fleet.User{ID: 1337},
+			false,
+			nil,
+			nil,
+			nil,
+			authz.ForbiddenErrorMessage,
+		},
+		{
+			"team id with free license",
+			&fleet.User{GlobalRole: ptr.String(fleet.RoleAdmin)},
+			false,
+			ptr.Uint(1),
+			nil,
+			nil,
+			ErrMissingLicense.Error(),
+		},
+		{
+			"team name with free license",
+			&fleet.User{GlobalRole: ptr.String(fleet.RoleAdmin)},
+			false,
+			nil,
+			ptr.String("team"),
+			nil,
+			ErrMissingLicense.Error(),
+		},
+		{
+			"team id and name specified",
+			&fleet.User{GlobalRole: ptr.String(fleet.RoleAdmin)},
+			true,
+			ptr.Uint(1),
+			ptr.String("team"),
+			nil,
+			"cannot specify both team_id and team_name",
+		},
+		{
+			"duplicate macOS profile name",
+			&fleet.User{GlobalRole: ptr.String(fleet.RoleAdmin)},
+			true,
+			ptr.Uint(1),
+			nil,
+			map[string][]byte{
+				"N1": mobileconfigForTest("N1", "I1"),
+				"N2": mobileconfigForTest("N1", "I2"),
+			},
+			`The name provided for the profile must match the profile PayloadDisplayName: "N1"`,
+		},
+		{
+			"duplicate macOS profile identifier",
+			&fleet.User{GlobalRole: ptr.String(fleet.RoleAdmin)},
+			true,
+			ptr.Uint(1),
+			nil,
+			map[string][]byte{
+				"N1": mobileconfigForTest("N1", "I1"),
+				"N2": mobileconfigForTest("N2", "I2"),
+				"N3": mobileconfigForTest("N3", "I1"),
+			},
+			`More than one configuration profile have the same identifier (PayloadIdentifier): "I1"`,
+		},
+		{
+			"only macOS",
+			&fleet.User{GlobalRole: ptr.String(fleet.RoleAdmin)},
+			false,
+			nil,
+			nil,
+			map[string][]byte{
+				"N1": mobileconfigForTest("N1", "I1"),
+				"N2": mobileconfigForTest("N2", "I2"),
+				"N3": mobileconfigForTest("N3", "I3"),
+			},
+			``,
+		},
+		{
+			"only windows",
+			&fleet.User{GlobalRole: ptr.String(fleet.RoleAdmin)},
+			false,
+			nil,
+			nil,
+			map[string][]byte{
+				"N1": syncMLForTest("./foo/bar"),
+				"N2": syncMLForTest("./baz"),
+				"N3": syncMLForTest("./zab"),
+				"N4": mobileconfigForTest("N4", "I1"),
+				"N5": mobileconfigForTest("N5", "I2"),
+				"N6": mobileconfigForTest("N6", "I3"),
+			},
+			``,
+		},
+		{
+			"mixed profiles",
```

Comment:
This one is the Windows only.

Reply:
thank you! I had this and the one above switched 😓